backend/src/apiserver/storage/job_store.go

```diff
@@ -112,15 +112,13 @@ func (s *JobStore) buildSelectJobsQuery(selectCount bool, opts *list.Options,
 	}
 
 	sqlBuilder := opts.AddFilterToSelect(filteredSelectBuilder)
-	if err != nil {
-		return "", nil, util.NewInternalServerError(err, "Failed to list jobs: %v", err)
-	}
 
 	// If we're not just counting, then also add select columns and perform a left join
 	// to get resource reference information. Also add pagination.
 	if !selectCount {
+		sqlBuilder = opts.AddPaginationToSelect(filteredSelectBuilder)
 		sqlBuilder = s.addResourceReferences(sqlBuilder)
-		sqlBuilder = opts.AddPaginationToSelect(sqlBuilder)
+		sqlBuilder = opts.AddSortingToSelect(sqlBuilder)
```

Comment:
Just wonder if it's still required to add sorting query if inner result has already been sorted.